kubeflow/trainer/rhai/transformers.py

```
@@ -475,28 +530,118 @@ def relative_update(self, inc=1):
+                    # Only print if we had checkpoints to check (not FileNotFoundError case)
+                    if checkpoint_dirs:
+                        print(
+                            "[Kubeflow] No valid checkpoints found in cloud storage. "
+                            "Training will start from scratch.",
+                            flush=True,
+                        )
+
+            # Barrier to wait for local rank 0 checkpoint download to complete
+            self._wait_for_all_ranks("download")
+
+        def on_save(self, args, state, control, **kwargs):
```

Ruff (0.14.14)

[warning] Unused method argument: `control`

(ARG002)